### PR TITLE
Add NullAway integration tests

### PR DESCRIPTION
**Pull request by theron-wang**
This PR adds the following NullAway integration tests to the evaluation script:

- na-89
- na-97 (pass)
- na-102 (pass)
- na-103 (pass)
- na-176
- na-323
- na-347 (pass)
- na-389 (pass)
- na-705
- na-791a (pass)
- na-791b

Most of these currently do not work as expected, so we should avoid forking these repos into the organization for now, just in case we need to modify `build.gradle` or other specimin files. Once we address the issues, we can work on the status json files in `src/main/resources`.

### CODE REVIEW

**Comment by kelloggm** [comment]
@theron-wang I'd like to merge this PR, so that we can get reliable results on CI for the NullAway bugs. In particular, I want to make sure that as we make changes to fix one bug, we don't accidentally break another that we already thought was working.

So, my plan is going to be to merge this PR as-is soon (unless you object), and then merge https://github.com/njit-jerse/specimin/pull/342 as soon as I can get the status right so that it passes CI.

**Review by kelloggm — APPROVED** [review]

**Comment by theron-wang** [comment]
@kelloggm Yep, sounds good. Let me just remove some debugging code before you merge. I also need to update the compilation scripts for na-97.

**Comment by theron-wang** [comment]
@kelloggm Should be ready to merge. I'm not sure if my change in `check_compilation.sh` is valid, though, so if you could check that I would appreciate it. Thanks!

**Comment by kelloggm** [comment]
> @kelloggm Should be ready to merge. I'm not sure if my change in check_compilation.sh is valid, though, so if you could check that I would appreciate it. Thanks!

Thanks. The change looks fine to me, but the best way to tell will just be to merge all of this and then see whether CI succeeds.

**Pull request merged**
